model/trigger/task.go

```diff
@@ -374,52 +371,51 @@ func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.No
 }
 
-func (t *taskTriggers) taskExceedsTime(sub *event.Subscription) (*notification.Notification, error) {
-	if sub == nil || sub.TriggerData == nil {
-		return nil, fmt.Errorf("subscription %s is missing trigger data", sub.ID)
-	}
-	thresholdString, ok := sub.TriggerData[taskDurationKey]
+func (t *taskTriggers) taskExceedsDuration(sub *event.Subscription) (*notification.Notification, error) {
+
+	thresholdString, ok := sub.TriggerData[event.TaskDurationKey]
 	if !ok {
 		return nil, fmt.Errorf("subscription %s has no task time threshold", sub.ID)
 	}
 	threshold, err := strconv.Atoi(thresholdString)
 	if err != nil {
 		return nil, fmt.Errorf("subscription %s has an invalid time threshold", sub.ID)
 	}
-	if t.event.EventType == event.TaskFinished {
-		maxDuration := time.Duration(threshold) * time.Second
-		if t.task.StartTime.Add(maxDuration).Before(t.task.FinishTime) {
-			return t.generate(sub, fmt.Sprintf("exceeded %d seconds", threshold))
-		}
+
+	maxDuration := time.Duration(threshold) * time.Second
+	if !t.task.StartTime.Add(maxDuration).Before(t.task.FinishTime) {
+		return nil, nil
 	}
-	return nil, nil
+	return t.generate(sub, fmt.Sprintf("exceeded %d seconds", threshold))
 }
 
-func (t *taskTriggers) taskRuntimeIncrease(sub *event.Subscription) (*notification.Notification, error) {
-	if sub == nil || sub.TriggerData == nil {
-		return nil, fmt.Errorf("subscription %s is missing trigger data", sub.ID)
-	}
-	percentString, ok := sub.TriggerData[percentIncreaseKey]
+func (t *taskTriggers) taskRuntimeChange(sub *event.Subscription) (*notification.Notification, error) {
+
+	percentString, ok := sub.TriggerData[event.TaskPercentChangeKey]
 	if !ok {
 		return nil, fmt.Errorf("subscription %s has no percentage increase", sub.ID)
 	}
-	percent, err := strconv.ParseFloat(percentString, 32)
+	percent, err := strconv.ParseFloat(percentString, 64)
 	if err != nil {
 		return nil, fmt.Errorf("subscription %s has an invalid percentage", sub.ID)
 	}
 
-	if t.event.EventType == event.TaskFinished {
-		lastGreen, err := t.task.PreviousCompletedTask(t.task.Project, []string{evergreen.TaskSucceeded})
-		if err != nil {
-			return nil, errors.Wrap(err, "error retrieving last green task")
-		}
-		thisTaskDuration := float64(t.task.FinishTime.Sub(t.task.StartTime))
-		prevTaskDuration := float64(lastGreen.FinishTime.Sub(lastGreen.StartTime))
-		ratio := thisTaskDuration / prevTaskDuration
-		percentIncrease := 100*ratio - 100
-		if percentIncrease > percent {
-			return t.generate(sub, fmt.Sprintf("increased in runtime by over %f%%", percent))
-		}
+	lastGreen, err := t.task.PreviousCompletedTask(t.task.Project, []string{evergreen.TaskSucceeded})
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving last green task")
+	}
+	if lastGreen == nil {
+		return nil, nil
+	}
+	thisTaskDuration := float64(t.task.FinishTime.Sub(t.task.StartTime))
+	prevTaskDuration := float64(lastGreen.FinishTime.Sub(lastGreen.StartTime))
+	if prevTaskDuration == 0 {
```

Comment:
Please guard prevTaskDuration and thisTaskDuration against Nan, +Inf and -Inf as described earlier. A util function is probably a good idea, `util.FloatIsNumeric`.

Reply:
Since these are cast from int64s, I don't think that's possible

Reply:
You're right, this should have referred to percentChange. If you do this, you also don't need the == 0 check since they're floats

Reply:
fixed